module/data/actor/group.mjs

```diff
@@ -0,0 +1,108 @@
+
+/**
+ * A data model and API layer which handles the schema and functionality of "group" type Actors in the dnd5e system.
+ *
+ * @example Create a new Group
+ * const g = new dnd5e.documents.Actor5e({
+ *  type: "group",
+ *  name: "Test Group",
+ *  system: {
+ *    members: ["3f3hoYFWUgDqBP4U"]
+ *  }
+ * });
+ */
+export default class GroupActor extends foundry.abstract.DataModel {
+
+  /** @override */
+  static defineSchema() {
+    const fields = foundry.data.fields;
+    return {
+      description: new fields.SchemaField({
+        full: new fields.HTMLField(),
+        summary: new fields.HTMLField()
+      }),
+      members: new fields.SetField(new fields.ForeignDocumentField(foundry.documents.BaseActor, {idOnly: true})),
+      attributes: new fields.SchemaField({
+        movement: new fields.SchemaField({
+          land: new fields.NumberField({initial: 0, nullable: false, integer: true, min: 0}),
+          water: new fields.NumberField({initial: 0, nullable: false, integer: true, min: 0}),
+          air: new fields.NumberField({initial: 0, nullable: false, integer: true, min: 0})
+        })
+      }),
+      currency: new fields.SchemaField({
+        pp: new fields.NumberField({nullable: false, initial: 0, integer: true, min: 0}),
+        gp: new fields.NumberField({nullable: false, initial: 0, integer: true, min: 0}),
+        ep: new fields.NumberField({nullable: false, initial: 0, integer: true, min: 0}),
+        sp: new fields.NumberField({nullable: false, initial: 0, integer: true, min: 0}),
+        cp: new fields.NumberField({nullable: false, initial: 0, integer: true, min: 0})
+      })
+    }
+  }
+
+  /* -------------------------------------------- */
+  /*  Data Preparation                            */
+  /* -------------------------------------------- */
+
+  /**
+   * Prepare base data for group actors.
+   * @internal
```

Comment:
`@internal` is a TSDoc code, not JSDoc so our linter isn't a fan.

```suggestion
   * @private
```

```suggestion
   * @protected
```

Reply:
We should change our linter then, because as of V10 our docstring syntax that we use is typedoc. We should use `@internal` tags for methods which are not intended for the public API but do need to be called outside of the scope in which they are defined.